### ISSUE TITLE
Reading invalid formatted plifile should raise error instead of warning

### ISSUE THREAD

**Issue by veenstrajelmer** [Deltares/HYDROLIB-core#320]
**Describe the bug**
Reading invalid formatted plifile should raise error instead of warning.

**To Reproduce**
```
polyfile_object = read_polyfile(file_pli,has_z_values=False)
```

**Expected behavior**
Case with comments added (not supported by hydrolib, but no issue for dflowfm kernel). I would say a # is necessary but also that is not supported.
```
DCSM-FM_OB_all_20181108
    5  2
-9.250000000000000E+000  4.300000000000000E+001 DCSM-FM_OB_all_20181108_0001
-9.500000000000000E+000  4.300000000000000E+001 DCSM-FM_OB_all_20181108_0002
-9.750000000000000E+000  4.300000000000000E+001 DCSM-FM_OB_all_20181108_0003
-1.000000000000000E+001  4.300000000000000E+001 DCSM-FM_OB_all_20181108_0004
-1.025000000000000E+001  4.300000000000000E+001 DCSM-FM_OB_all_20181108_0005
```
>> UserWarning: Expected a valid next point at line 2.
>> should raise exception of invalid formatted plifile

Case with invalid nrows in plifile
```
DCSM-FM_OB_all_20181108
    206    2
-9.250000000000000E+000  4.300000000000000E+001 
-9.500000000000000E+000  4.300000000000000E+001 
-9.750000000000000E+000  4.300000000000000E+001 
-1.000000000000000E+001  4.300000000000000E+001 
-1.025000000000000E+001  4.300000000000000E+001 
```
>> UserWarning: EoF encountered before the block is finished.
>> should raise exception of invalid formatted plifile

Case with valid formatted pli file
```
DCSM-FM_OB_all_20181108
    5   2
-9.250000000000000E+000  4.300000000000000E+001 
-9.500000000000000E+000  4.300000000000000E+001 
-9.750000000000000E+000  4.300000000000000E+001 
-1.000000000000000E+001  4.300000000000000E+001 
-1.025000000000000E+001  4.300000000000000E+001 
```
>> UserWarning: White space at the start of the line is ignored.
>> this is not to relevant to warn about, it is about the white space before '5'. Disable warning?
for the last thing, I created a separate issue: https://github.com/Deltares/HYDROLIB-core/issues/370

**Version info (please complete the following information):**
 - OS: Windows, Anaconda, Spyder
 - Version: 0.3.0

**Comment by veenstrajelmer** [comment]
Would be nice if the last point can be resolved before 0.4, maybe as part of https://github.com/Deltares/HYDROLIB-core/issues/34 ?

**Comment by MRVermeulenDeltares** [comment]
Case with valid formatted pli file & UserWarning: White space at the start of the line is ignored, already resolved in #370

**Comment by MRVermeulenDeltares** [comment]
In all cases in which the file is invalid, an exception error should be raised.
